tempesta_fw/http.c

```diff
@@ -2930,6 +2934,37 @@ tfw_http_req_process(TfwConn *conn, const TfwFsmData *data)
 		skb = NULL;
 	}
 
+	/*
+	 * If a client work through a forward proxy, then a proxy can pass it's
+	 * IP address by the first value in X-Forwarded-For
+	 */
+	s_xff = req->h_tbl->tbl[TFW_HTTP_HDR_X_FORWARDED_FOR];
+	s_ip = tfw_str_next_str_val(&s_xff);
+	nchunks = 0;
+	TFW_STR_FOR_EACH_CHUNK(c, &s_ip, end) {
+		if (!(c->flags & TFW_STR_VALUE))
+			break;
+		nchunks++;
+	}
+	s_ip.nchunks = nchunks;
```

Comment:
`tfw_http_req_process()` is too long and overloaded now, so it's better to move the str logic to a small helping function. I think the whole new code with the new local variables can and should be moved to the helper.